spec/models/reservation_spec.rb

```diff
@@ -1,725 +1,913 @@
+      expect(res.reserver).to be_new_record
     end
   end
 
-  context 'when in a final status' do
-    subject(:reservation) do
-      FactoryGirl.create(:valid_reservation)
-    end
-    it 'the status should not be able to be changed if denied' do
-      reservation.update_attributes(status: 'denied')
-      reservation.status = 'reserved'
-      expect { reservation.save! }.to raise_error ActiveRecord::RecordInvalid
-    end
-    it 'the status should not be able to be changed if missed' do
-      reservation.update_attributes(
-        FactoryGirl.attributes_for(:missed_reservation)
-      )
-      reservation.status = 'reserved'
-      expect { reservation.save! }.to raise_error ActiveRecord::RecordInvalid
-    end
-    it 'the status should not be able to be changed if returned' do
-      reservation.update_attributes(
-        FactoryGirl.attributes_for(:checked_in_reservation)
-      )
-      reservation.status = 'reserved'
-      expect { reservation.save! }.to raise_error ActiveRecord::RecordInvalid
+  describe '#find_renewal_length' do
+    let!(:length) { 5 }
+    let!(:model) do
+      FactoryGirl.create(:equipment_model_with_item, max_renewal_length: length)
+    end
+    let!(:res) do
+      FactoryGirl.build_stubbed(:valid_reservation,
+                                reserver: FactoryGirl.create(:user),
+                                equipment_model: model)
+    end
+    it 'sets the correct renewal length' do
+      expect(res.find_renewal_date).to eq(res.due_date + length.days)
+    end
+    context 'with a blackout date overlapping with the max renewal length' do
+      it 'sets the correct renewal length' do
+        FactoryGirl.create(:blackout,
+                           start_date: res.due_date + 2.days,
+                           end_date: res.due_date + length.days + 1.day)
+        expect(res.find_renewal_date).to eq(res.due_date + 1.day)
+      end
+    end
+    context 'with a blackout date going right up to the max renewal length' do
+      it 'sets a length of 0' do
+        FactoryGirl.create(:blackout,
+                           start_date: res.due_date + 1.day,
+                           end_date: res.due_date + length.days + 1.day)
+        expect(res.find_renewal_date).to eq(res.due_date)
+      end
+    end
+    context 'with a future reservation on the same model' do
+      it 'sets the correct renewal length' do
+        FactoryGirl.create(:reservation,
+                           equipment_model: model,
+                           start_date: res.due_date + 3.days,
+                           due_date: res.due_date + length.days + 1.day)
+        expect(res.find_renewal_date).to eq(res.due_date + 2.days)
+      end
     end
   end
 
-  context '#approved?' do
-    subject(:reservation) { FactoryGirl.create(:request) }
-
-    it 'returns false if requested' do
-      expect(reservation.approved?).to be_falsey
+  describe '#eligible_for_renew?' do
```

Comment:
Ahaha that's kinda funny. So are we going to check for it here as well and test that?